### DISCUSSION TITLE
Named bookmarks not recognized on Windows Powershell

### DISCUSSION THREAD

**Discussion by abrahambahez** [jotaen/klog#371] [Bugs]
Just upgrade klog, excited for the multiple bookmarks feature. But when I try to set a new one, it just assign it to `default`.

I've tried in Linux and work well

![image](https://user-images.githubusercontent.com/53906659/136669541-eb89c5a1-9edf-421d-b01f-ef73ae32511b.png)

**Comment by vladdeSV** [comment]
This is related to you using PowerShell.

Try quoting the bookmark name:

```
klog bookmarks set D:\vidadigital\pd\doctorado.klg '@doctorado'
```

**Comment by vladdeSV** [comment]
@jotaen Why are bookmark names required to be prefixed by an `@` when creating a bookmark? Unless there is some specific reason for this, then maybe this requirement can be removed to help PowerShell users?

Pseudo-code in the bookmark creation
```
if name[0] != '@'
  name = '@' + name
```

**Comment by jotaen** [comment]
@vladdeSV The prefix is not required when setting a bookmark, only when using it. So for setting you can say `klog bookmarks set file.klg foo` and then use it via `klog total @foo`.

I didn't test on Powershell unfortunately, only with cmd.exe. But I assume if setting doesn't work, then using the named bookmark won't probably work either on Powershell.

**Comment by vladdeSV** [comment]
Ah, that is true.

Best suggestion I have then would be to add a note in the documentation that PowerShell users need to quote the bookmark name.

**Comment by jotaen** [comment]
Yes, even though quoting is inconvenient, it might still be the best workaround... I won't be able to look into this more this week, so I'll take care of it later. I hope it can be fixed somehow, otherwise I'll add some warning / hint in the docs. Thanks for reporting!

**Comment by vladdeSV** [comment]
This issue should be closed, or possibly renamed to reflect that this is PowerShell related.

Solution to problem is to quote the bookmark name (`'@myBookmark'`).

**Comment by jotaen** [comment]
The `@` operator is special syntax on [Windows Powershell for hash tables or “splatting”](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_splatting). It is resolved *before* klog is invoked, so unfortunately it’s not possible for klog to detect or handle it.

I’ve added a [note in the documentation](https://klog.jotaen.net/#bookmarks) how to workaround this. You can either prepend the bookmark name with a backtick, or you wrap it in quotes. So both the following ways will do:

```
klog total `@foo
klog total '@foo'
```
